keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,789 @@
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+We manually test it in our internal environment and it works. It's worth doing automated upgrade/rollback tests in the future.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Examine the kubelet configuration of a node to verify the existence of the feature gate and the utilization of the new policy option.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [x] Other (treat as last resort)
+  - Details: Inspect the kubelet configuration of the nodes: check feature gate and usage of the new option.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+It won't cause any performance regression. So we don't need to define any SLOs for this feature.
```

Comment:
arguably a value too high CAN cause performance regressions in the admission flow.
OTOH in the current state of things it doesn't work at all which is probably a even worse performance?
I think we should somehow still mention the performance implication though

Reply:
> arguably a value too high CAN cause performance regressions in the admission flow

Could you explain more details about this? thanks.

Reply:
The reason we add 8 is because the internal current topology manager logic need to compute states using an algorithm which will create many states. I think is something along the lines of `O(N**2)` if I'm not mistaken? Please don't quote me, need to check. However , the algo it is surely super-linear so the processing time of  the topology manager decision changes accordingly.
Now, Topology Manager runs in the kubelet admission flow. Users should be aware that if they set `max-allowable-numa-nodes` say 16, then Topology Manager it won't run roughly 2 times slower, but much slower.

Reply:
thanks for the detail!